sqlglot/optimizer/simplify.py

```diff
@@ -257,6 +257,12 @@ def _simplify_binary(expression, a, b):
                 return TRUE if not_ else FALSE
             if a == NULL:
                 return FALSE if not_ else TRUE
+    elif isinstance(expression, exp.NullSafeEQ):
+        if a == b:
```

Comment:
is this correct? how do you know if column a == column b? you don't know the values

Reply:
I was following the same logic as the EQ case below it except NULL <=> NULL is TRUE so I put it before the null check
```
    if isinstance(expression, exp.EQ) and a == b:
        return TRUE
```

Reply:
brain fart, this looks correct, because a == b means that the column is the same